### ISSUE TITLE
Handling of local transactions in txpool

### ISSUE THREAD

**Issue by MariusVanDerWijden** [ethereum/go-ethereum#21242]
This is a reminder issue about a low priority refactor we might attempt in the future.

In our txpool implementation we differentiate between local and non-local transaction. Local transactions will not be discarded from the txpool while non-local transactions can. This has the disadvantage that if we want to add a new local transaction to a full txpool, we have to linearly search the entire txpool for transactions that can be discarded. If the txpool is full of local transactions (e.g. if we run a stress test on a local node) this `(txPricedList).Discard()` takes a long time. This has also impact on other parts of the system. Because our miner depends on the same lock that is locks the txpool, the miner cannot proceed which only congests the node further. 
This problem however is only relevant for nodes with a lot of  local transaction and also mines. 

Some ideas:
- create a second mutex so we can unlock the pool mutex, s.th. the miner does not wait on the mutex
- split the txpools for local and non-local transactions completely
- add a watchdog that collects local transactions to insert them later

As previously stated, this issue is mostly irrelevant for real-life use cases and is only really prevalent in synthetic benchmarks.

**Comment by MariusVanDerWijden** [comment]
closed in favor of https://github.com/ethereum/go-ethereum/issues/21174
